src/utils/api_helper.ts

```diff
@@ -1164,3 +1234,94 @@ export const getNewestCoupon = async (
     { signal }
   )
 }
+
+export const findValidSlotForGuestScheduling = async (
+  meetingId: string
+): Promise<string | null> => {
+  try {
+    // Get the conference meeting using the conference endpoint
+    const conferenceMeeting = await getConferenceMeeting(meetingId)
+    if (!conferenceMeeting) {
+      console.error(`No conference meeting found for meeting ${meetingId}`)
+      return null
+    }
+
+    // Get the list of slot IDs from the meeting
+    const slotIds = conferenceMeeting.slots || []
+    if (slotIds.length === 0) {
+      console.error(
+        `No slots found for conference meeting ${conferenceMeeting.id}`
+      )
+      return null
+    }
+
+    // Check which of these slot IDs actually exist in the slots table
+    const existingSlots = await getSlotsByIds(slotIds)
+    if (!existingSlots || existingSlots.length === 0) {
+      console.error(
+        `No existing slots found for conference meeting ${conferenceMeeting.id}`
+      )
+      return null
+    }
+
+    // Return the first valid slot ID, this is the slot ID that will be used for the guest's reschedule link
+    return existingSlots[0]?.id || null
+  } catch (error) {
+    console.error(
+      `Error finding valid slot for guest scheduling ${meetingId}:`,
+      error
+    )
+    return null
+  }
+}
+
+export const sendGuestNotification = async (
```

Comment:
We should never expose the endpoint for sending notifications that's something that can be exploited fast.

Reply:
Notifications should only be server sent